### ISSUE TITLE
Flipper 0.126 crashing metro

### ISSUE THREAD

**Issue by artstylee** [facebook/flipper#3189]
## 🐛 Bug Report

after installing flipper 0.126 my metro crashing every time with an error

```
D:\docs\proj\react-express-jwt-full\motoNative\node_modules\metro-he
rmes-compiler\src\emhermesc.js:77
          throw ex;
          ^

TypeError: Cannot read properties of undefined (reading 'startsWith'
)

```

## Environment

WIN 10

**Comment by xD3CODER** [comment]
Did you unzip the files over the old installation? If so, try removing and reinstalling Flipper completely, that solved the problem for me

**Comment by mweststrate** [comment]
This should be fixable by changing `node_modules/@react-native-community/cli-server-api/build/websocket/eventSocketServer.js` line 131 from ` return origin.startsWith('http://localhost:') || origin.startsWith('file:');` to `return !origin || origin.startsWith('http://localhost:') || origin.startsWith('file:');`.

I'll try to submit a PR for that when back from PTO, or feel free to submit a PR already at https://github.com/react-native-community/cli/blob/master/packages/cli-server-api/src/websocket/eventsSocketServer.ts#L129. You can solve the issue locally by using `patch-package`

**Comment by davydspotnana** [comment]
Faced with this issue too, downgrading Flipper to 0.125 will fix it

**Comment by realtril** [comment]
@davydspotnana how did you downgrade a flipper?

**Comment by artstylee** [comment]
ye i downgrade too

**Comment by realtril** [comment]
@artstylee how did you downgrade a flipper? I mean, how can I downgrade desctop app?

**Comment by sytolk** [comment]
the same issue on Mac
```
info Opening flipper://null/Hermesdebuggerrn?device=React%20Native...
/Users/sytolk/IdeaProjects/drivefuel-app/DriveApp/node_modules/metro-hermes-compiler/src/emhermesc.js:77
          throw ex;
          ^

TypeError: Cannot read property 'startsWith' of undefined
    at Object.verifyClient (/Users/sytolk/IdeaProjects/drivefuel-app/DriveApp/node_modules/@react-native-community/cli-server-api/build/websocket/eventsSocketServer.js:131:21)
    at WebSocketServer.handleHybiUpgrade (/Users/sytolk/IdeaProjects/drivefuel-app/DriveApp/node_modules/ws/lib/WebSocketServer.js:331:28)
    at WebSocketServer.handleUpgrade (/Users/sytolk/IdeaProjects/drivefuel-app/DriveApp/node_modules/ws/lib/WebSocketServer.js:173:26)
    at Server.WebSocketServer._onServerUpgrade (/Users/sytolk/IdeaProjects/drivefuel-app/DriveApp/node_modules/ws/lib/WebSocketServer.js:89:12)
    at Server.emit (events.js:412:35)
    at onParserExecuteCommon (_http_server.js:700:14)
    at onParserExecute (_http_server.js:613:3)
Process terminated. Press <enter> to close the window
```

**Comment by realtril** [comment]
@sytolk thank you!

**Comment by renchap** [comment]
@mweststrate I have been able to test your fix and it works.

I can make a PR, but I am not sure this is the best issue as this check is here for security reasons, and allowing empty origins could be an issue.

**Comment by nicode-io** [comment]
Thanks @mweststrate fix is working and no need to downgrade so.

**Comment by nicode-io** [comment]
Thanks @mweststrate ! Your solution works for me

**Comment by amirping** [comment]
+1

**Comment by ct-tomas-sykora** [comment]
Happens on MacOS with Apple Silicon as well.

**Comment by mweststrate** [comment]
Should be fixed in Flipper 0.127 which will ship later today, and/or in the next version of  RN

**Comment by fergalmoran** [comment]
I'm not sure this is a duplicate of #1523 ? I downgraded to 0.125.0 then upgraded to 0.127.0 but none of them fixed my issue.
Flipper connects to my device but the hermes plugin is disabled.
```Plugin 'Hermes Debugger (RN)' is unavailable \
Device plugin 'Hermes Debugger (RN)' is not supported by the selected device 'Pixel_5_API_30' (Android).```
